src/stirling/source_connectors/socket_tracer/testing/socket_trace_bpf_test_fixture.h

```diff
@@ -46,104 +48,31 @@ class SocketTraceBPFTestFixture : public ::testing::Test {
     // TODO(oazizi): Setting flags in the test is a bad idea because of the pitfall above.
     //               Change this paradigm.
     FLAGS_treat_loopback_as_in_cluster = !EnableClientSideTracing;
-
-    auto source_connector = SocketTraceConnector::Create("socket_trace_connector");
-
-    source_.reset(dynamic_cast<SocketTraceConnector*>(source_connector.release()));
-    ASSERT_OK(source_->Init());
-
-    source_->set_data_tables(data_tables_.tables());
-
-    // Cause Uprobes to deploy in a blocking manner.
-    // We don't return until the first set of uprobes has successfully deployed.
-    RefreshContext(/* blocking_deploy_uprobes */ true);
+    ASSERT_OK(source_.Init());
   }
 
-  void TearDown() override { ASSERT_OK(source_->Stop()); }
+  void TearDown() override { ASSERT_OK(source_.Stop()); }
 
   void ConfigureBPFCapture(traffic_protocol_t protocol, uint64_t role) {
-    auto* socket_trace_connector = dynamic_cast<SocketTraceConnector*>(source_.get());
+    auto* socket_trace_connector = dynamic_cast<SocketTraceConnector*>(source_.RawPtr());
     ASSERT_OK(socket_trace_connector->UpdateBPFProtocolTraceRole(protocol, role));
   }
 
   void TestOnlySetTargetPID(int64_t pid) {
     FLAGS_test_only_socket_trace_target_pid = pid;
-    auto* socket_trace_connector = dynamic_cast<SocketTraceConnector*>(source_.get());
+    auto* socket_trace_connector = dynamic_cast<SocketTraceConnector*>(source_.RawPtr());
     ASSERT_OK(socket_trace_connector->TestOnlySetTargetPID());
   }
 
-  void RefreshContext(bool blocking_deploy_uprobes = false) {
-    absl::base_internal::SpinLockHolder lock(&socket_tracer_state_lock_);
-
-    RefreshContextCore();
-
-    if (blocking_deploy_uprobes) {
-      source_->InitContext(ctx_.get());
-    }
-  }
-
-  void StartTransferDataThread() {
-    // Drain the perf buffers before starting the thread.
-    // Otherwise, perf buffers may already be full, causing lost events and flaky test results.
-    source_->PollPerfBuffers();
-
-    transfer_data_thread_ = std::thread([this]() {
-      transfer_enable_ = true;
-      while (transfer_enable_) {
-        {
-          absl::base_internal::SpinLockHolder lock(&socket_tracer_state_lock_);
-          RefreshContextCore();
-          source_->TransferData(ctx_.get());
-        }
-        std::this_thread::sleep_for(kTransferDataPeriod);
-      }
-    });
-
-    while (!transfer_enable_) {
-    }
-
-    // Wait for at least one TransferData() call before returning.
-    std::this_thread::sleep_for(kTransferDataPeriod);
-  }
-
-  void StopTransferDataThread() {
-    // Give enough time for one more TransferData call by transfer_data_thread_,
-    // so we make sure we've captured everything.
-    std::this_thread::sleep_for(2 * kTransferDataPeriod);
-
-    absl::base_internal::SpinLockHolder lock(&socket_tracer_state_lock_);
-    CHECK(transfer_data_thread_.joinable());
-    transfer_enable_ = false;
-    transfer_data_thread_.join();
-  }
```

Comment:
Is there a reason why these methods weren't reimplemented `UnitConnector` style? It seems like doing so would reduce the number of changes in all of the test files.